File: 카카오/2019 카카오 개발자 겨울 인턴십/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    vector<int> lines ;
    int last = 0;
    
    for (int i = 0; i < moves.size(); i++){
        last = 0;
        
        for(int j = 0; j < board.size(); j++){
            if(board[j][moves[i]-1] == 0){}
            else {
                last = board[j][moves[i]-1];
                board[j][moves[i]-1] = 0;
                break;
            }
        }
        if(last ==0)continue;
       
        if(lines.size() == 0)lines.push_back(last);
        else if (lines.back() == last){
            answer += 2;
            lines.pop_back();
        }
        else{
            lines.push_back(last);
        }
        
        

    }
    
    
    
    return answer;
}

File: 카카오/2019 카카오 개발자 겨울 인턴십/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    result = []
    for i in moves:
        for j in range(len(board)):
            if board[j][i - 1] == 0:
                continue
            else:
                if len(result) == 0:
                    result.append(board[j][i - 1])
                    board[j][i - 1] = 0
                    break
                
                elif result[-1] == board[j][i - 1]:
                    result.pop()
                    board[j][i - 1] = 0
                    answer += 2
                    break
                
                else:
                    result.append(board[j][i - 1])
                    board[j][i - 1] = 0
                    break
    return answer
